src/olympia/scanners/views.py

```diff
@@ -37,21 +60,39 @@ def get_queryset(self):
             .all()
         )
         bad_results = (
-            ScannerResult.objects.exclude(version=None)
-            .filter(state=TRUE_POSITIVE)
+            bad_results.filter(state=TRUE_POSITIVE)
             .annotate(label=Value(LABEL_BAD, output_field=CharField()))
             .all()
         )
-        return good_results.union(bad_results).order_by('-created')
+
+        queryset = ScannerResult.objects.none()
+
+        if not label:
+            queryset = good_results.union(bad_results)
+        elif label == LABEL_GOOD:
+            queryset = good_results
+        elif label == LABEL_BAD:
```

Comment:
So if there is a `label` and it's neither `LABEL_GOOD` nor `LABEL_BAD` we return an empty queryset? Is that the behaviour we want?

Reply:
Oh, wait, I see we throw an error on `get` if the `label` is invalid, but I guess you want this behaviour as a fallback in case a `queryset` is somehow requested outside of a `get`?

Reply:
If the label is neither GOOD nor BAD, we shouldn't return anything indeed. It is less useful now that we have some checks in the `get` method but at least it will always return a queryset.